src/types/hash-160.ts

```diff
@@ -7,7 +7,11 @@ class Hash160 extends Hash {
   static readonly width = 20;
   static readonly ZERO_160: Hash160 = new Hash160(Buffer.alloc(Hash160.width));
 
-  constructor(bytes: Buffer) {
+  constructor(bytes?: Buffer) {
+    if (bytes && bytes.byteLength === 0) {
+      bytes = Hash160.ZERO_160.bytes;
+    }
+
     super(bytes ?? Hash160.ZERO_160.bytes);
```

Comment:
I think you can remove the ?? since bytes can never be undefined now?

Reply:
No, the empty buffer should be handled differently than an undefined parameter.

Reply:
Agree - misread.